lib/MusicBrainz/Server/Controller/WS/js.pm

```diff
@@ -802,13 +805,32 @@ sub get_json_request_body : Private {
     return $decoded_object;
 }
 
-sub check_login : Private {
+sub cookie_login_or_error : Private {
     my ($self, $c, $error) = @_;
 
     $c->forward('/user/cookie_login') unless $c->user_exists;
     $self->detach_with_error($c, $error) unless $c->user_exists;
 }
 
+sub check_login : Chained('root') PathPart('check-login') {
+    my ($self, $c) = @_;
+
+    $c->forward('/user/cookie_login');
```

Comment:
Is checking `$c->user_exists` unneeded here because it is checked within `cookie_login` anyway?
Should this check be skipped too for the other `cookie_login` call in the above `cookie_login_or_error`?

Reply:
Yeah, it's redundant there too; I can remove it in a separate commit.